server/src/internalClusterTest/java/org/elasticsearch/snapshots/SharedClusterSnapshotRestoreIT.java

```diff
@@ -2044,7 +2044,8 @@ public void testSnapshotDeleteRelocatingPrimaryIndex() throws Exception {
         final String indexName = "test-idx";
         assertAcked(prepareCreate(indexName, 2, indexSettingsNoReplicas(between(2, 10))));
         ensureGreen(indexName);
-        indexRandomDocs(indexName, 100);
+        // Use a large number of documents so the relocation below takes long enough for the snapshot to run concurrently
+        indexRandomDocs(indexName, 25000);
```

Comment:
Can we instead block relocation from completing? Otherwise there's still theoretically a race (albeit less likely)

Reply:
Do we have a pre-packaged way of doing that in tests? (I couldn't find one, hence this fix) I could try to come up with one if we don't but it looks like that would take a bit of work.

Reply:
Something along the lines of `IndexRecoveryIT.RecoveryActionBlocker`

Reply:
Thanks! Could've though of that myself :facepalm: -> moved the test to the dedicated cluster tests now for this so I can just black-hole the file chunk requests but don't have to bother with cleaning up the cluster networking after the test.